File: frontend/src/api/fleetHealth.ts
// frontend/src/api/fleetHealth.ts
import { api } from './client'

export interface NodeHealthSnapshot {
  id: string
  node_id: string
  minion_id: string
  hostname: string | null
  collected_at: string
  disk_root_used_gb: number | null
  disk_root_total_gb: number | null
  disk_root_pct: number | null
  disk_root_inodes_pct: number | null
  mem_total_gb: number | null
  mem_available_gb: number | null
  mem_used_pct: number | null
  cpu_load_1m: number | null
  cpu_load_5m: number | null
  cpu_load_15m: number | null
  uptime_seconds: number | null
  gpu_name: string | null
  gpu_vram_mb: number | null
  cpu_power_mw: number | null
  gpu_power_mw: number | null
  thermal_pressure: string | null
  error: string | null
  // computed by API
  disk_alert: boolean
  mem_alert: boolean
  thermal_alert: boolean
}

export const fleetHealthApi = {
  getFleetHealth: () => api.get<NodeHealthSnapshot[]>('/api/v1/fleet-health'),
  triggerCollect: () => api.post<{ status: string; message: string }>('/api/v1/fleet-health/collect', {}),
  getNodeHistory: (nodeId: string, hours = 24) =>
    api.get<NodeHealthSnapshot[]>(`/api/v1/fleet-health/${nodeId}/history?hours=${hours}`),
}

export function formatUptime(seconds: number | null): string {
  if (seconds === null) return '—'
  const d = Math.floor(seconds / 86400)
  const h = Math.floor((seconds % 86400) / 3600)
  const m = Math.floor((seconds % 3600) / 60)
  if (d > 0) return `${d}d ${h}h`
  if (h > 0) return `${h}h ${m}m`
  return `${m}m`
}

export function formatPower(mw: number | null): string {
  if (mw === null) return '—'
  return mw >= 1000 ? `${(mw / 1000).toFixed(1)} W` : `${mw} mW`
}

export function thermalColor(pressure: string | null): string {
  switch (pressure) {
    case 'Nominal': return 'text-green-600'
    case 'Light': return 'text-yellow-500'
    case 'Moderate': return 'text-orange-500'
    case 'Heavy': return 'text-red-600'
    case 'Critical': return 'text-red-800 font-bold'
    default: return 'text-gray-400'
  }
}

File: fleet_platform/schemas/fleet_health.py
# fleet_platform/schemas/fleet_health.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, computed_field


class NodeHealthSnapshotResponse(BaseModel):
    id: uuid.UUID
    node_id: uuid.UUID
    minion_id: str
    hostname: str | None
    collected_at: datetime
    disk_root_used_gb: Decimal | None
    disk_root_total_gb: Decimal | None
    disk_root_pct: int | None
    disk_root_inodes_pct: int | None
    mem_total_gb: Decimal | None
    mem_available_gb: Decimal | None
    mem_used_pct: int | None
    cpu_load_1m: Decimal | None
    cpu_load_5m: Decimal | None
    cpu_load_15m: Decimal | None
    uptime_seconds: int | None
    gpu_name: str | None
    gpu_vram_mb: int | None
    cpu_power_mw: int | None
    gpu_power_mw: int | None
    thermal_pressure: str | None
    error: str | None

    @computed_field  # type: ignore[misc]
    @property
    def disk_alert(self) -> bool:
        return (self.disk_root_pct or 0) >= 85

    @computed_field  # type: ignore[misc]
    @property
    def mem_alert(self) -> bool:
        return (self.mem_used_pct or 0) >= 90

    @computed_field  # type: ignore[misc]
    @property
    def thermal_alert(self) -> bool:
        return self.thermal_pressure not in (None, "Nominal")

    model_config = {"from_attributes": True}


class CollectResponse(BaseModel):
    status: str
    message: str
